resources/lambda/browsershot.js

```diff
@@ -47,8 +48,39 @@ exports.handle = async function (event) {
     // If there was a path, then read the file and return it.
     if (event.options.path) {
         let contents = fs.readFileSync(event.options.path);
+
         fs.unlinkSync(event.options.path);
 
+        // If the file destination is S3, then write
+        // the file and return the ETag as confimation.
+        if (event.options.s3) {
+            const s3 = new AWS.S3({ region: event.options.s3.region });
+
+            let type;
+
+            switch (event.options.type) {
+                case "png":
+                    type = 'image/png';
+                    break;
+                case "jpeg":
+                    type = 'image/jpeg';
+                    break;
+                default:
+                    type = 'application/pdf';
+            }
+
+            const params = {
+                Bucket: event.options.s3.bucket,
+                Key: event.options.s3.path,
+                Body: contents,
+                ContentType: type
+            }
+
+            const result = await s3.putObject(params).promise();
+
+            return result.ETag
```

Comment:
Would it make sense to return the whole `result` here? Or do we need to document that the ETag is returned?

Reply:
From what I can see, 'ETag' is the only value in the result object. I did originally have the whole object returned, but Spatie's change this morning means that `callBrowser` has to return a string.

We could add more logic in the custom `callBrowser` method to analyse the result and then return some other string, but I ultimately decided it was arbitrary as we just need some confirmation that an S3 object was created.

Reply:
Thanks for the explanation. Makes sense to leave it as is.